Refactor the ll40ls namespace driver methods to more closely match other distance sensor driver implementations

mcsauder opened the pull request:
**Describe problem solved by the proposed pull request**
The LidarLite driver is somewhat non-standard at the moment.  This PR aligns the driver with other distance sensor drive implementations.

**Additional context**
Please see #12278 and #12690

mcsauder commented:
@cmic0 , I put inplace in thie PR your `LidarLiteI2C::probe()` simplifications here as well, nice work on that.

I've simplified and standardized this driver's namespace methods against the other i2c distance sensor drivers' methods and although the diff isn't beautiful, once all of the distance sensor drivers are  aligned it will be worth it.  (Back in March I mentioned in a dev call that I thought this would be a 6 month process... it's getting close now.)

@ar1040, would you mind giving this one a go? (Ensure your have `SENS_EN_LL40LS` set appropriately for your hardware.)  Thanks!

TSC21 commented:
@dagar Could we try to bring this PR first than #12756?

TSC21 commented:
@ar1040 On a separate dir: `git clone --recursive https://github.com/mcsauder/Firmware.git -b lidar_lite`. Then build and test.

mcsauder commented:
I'm not quite sure what to do about this, but I have hit it in about four of my open PR's so far today and my resolution has been to copy out my changes, destroy my branch, start over, and force push the changes... I'll do the same with this branch shortly:
```
mark@mark-laptop:~/code/Firmware$ git merge --continue 
boards/px4/sitl/src/CMakeLists.txt:39: new blank line at EOF.
platforms/CMakeLists.txt:35: new blank line at EOF.
platforms/common/apps.cpp.in:121: new blank line at EOF.
platforms/common/module.cpp:167: new blank line at EOF.
platforms/common/work_queue/dq_addlast.c:73: new blank line at EOF.
platforms/common/work_queue/dq_rem.c:81: new blank line at EOF.
platforms/common/work_queue/dq_remfirst.c:81: new blank line at EOF.
platforms/common/work_queue/hrt_queue.c:140: new blank line at EOF.
platforms/common/work_queue/hrt_thread.c:296: new blank line at EOF.
platforms/common/work_queue/queue.c:97: new blank line at EOF.
platforms/common/work_queue/sq_addlast.c:72: new blank line at EOF.
platforms/common/work_queue/sq_remfirst.c:76: new blank line at EOF.
platforms/nuttx/src/px4/CMakeLists.txt:37: new blank line at EOF.
platforms/nuttx/src/px4/common/include/px4_platform/adc.h:40: new blank line at EOF.
platforms/nuttx/src/px4/common/include/px4_platform/micro_hal.h:56: new blank line at EOF.
platforms/nuttx/src/px4/nxp/CMakeLists.txt:36: new blank line at EOF.
platforms/nuttx/src/px4/nxp/k66/CMakeLists.txt:40: new blank line at EOF.
platforms/nuttx/src/px4/nxp/k66/include/px4_arch/adc.h:36: new blank line at EOF.
platforms/nuttx/src/px4/nxp/k66/include/px4_arch/io_timer.h:37: new blank line at EOF.
platforms/nuttx/src/px4/nxp/kinetis/adc/adc.cpp:183: new blank line at EOF.
platforms/nuttx/src/px4/nxp/kinetis/hrt/CMakeLists.txt:38: new blank line at EOF.
platforms/nuttx/src/px4/nxp/kinetis/include/px4_arch/adc.h:40: new blank line at EOF.
platforms/nuttx/src/px4/stm/CMakeLists.txt:36: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32_common/adc/adc.cpp:230: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32_common/hrt/CMakeLists.txt:38: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32_common/include/px4_arch/adc.h:57: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32_common/include/px4_arch/px4io_serial.h:169: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32f1/include/px4_arch/io_timer.h:37: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32f1/include/px4_arch/micro_hal.h:51: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32f1/io_timer.h:37: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32f3/CMakeLists.txt:36: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32f3/include/px4_arch/adc.h:36: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32f3/include/px4_arch/io_timer.h:37: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32f3/include/px4_arch/micro_hal.h:51: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32f4/CMakeLists.txt:42: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32f4/include/px4_arch/adc.h:36: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32f4/include/px4_arch/io_timer.h:37: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32f4/include/px4_arch/micro_hal.h:51: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32f4/include/px4_arch/px4io_serial.h:38: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32f4/px4io_serial/px4io_serial.cpp:471: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32f7/CMakeLists.txt:42: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32f7/include/px4_arch/adc.h:36: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32f7/include/px4_arch/io_timer.h:37: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32f7/include/px4_arch/micro_hal.h:55: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32f7/include/px4_arch/px4io_serial.h:38: new blank line at EOF.
platforms/nuttx/src/px4/stm/stm32f7/px4io_serial/px4io_serial.cpp:502: new blank line at EOF.
platforms/posix/src/px4/CMakeLists.txt:37: new blank line at EOF.
platforms/posix/src/px4/common/CMakeLists.txt:79: new blank line at EOF.
platforms/posix/src/px4/common/include/px4_platform/micro_hal.h:34: new blank line at EOF.
platforms/posix/src/px4/common/px4_daemon/CMakeLists.txt:42: new blank line at EOF.
platforms/posix/src/px4/common/px4_daemon/client.h:88: new blank line at EOF.
platforms/posix/src/px4/common/px4_daemon/history.h:104: new blank line at EOF.
platforms/posix/src/px4/common/px4_daemon/pxh.h:98: new blank line at EOF.
platforms/posix/src/px4/common/px4_daemon/server.h:127: new blank line at EOF.
platforms/posix/src/px4/common/px4_daemon/sock_protocol.cpp:51: new blank line at EOF.
platforms/posix/src/px4/common/px4_daemon/sock_protocol.h:48: new blank line at EOF.
platforms/posix/src/px4/common/px4_posix_impl.cpp:101: new blank line at EOF.
platforms/posix/src/px4/common/px4_posix_tasks.cpp:452: new blank line at EOF.
platforms/posix/src/px4/generic/CMakeLists.txt:36: new blank line at EOF.
platforms/posix/src/px4/generic/generic/CMakeLists.txt:36: new blank line at EOF.
platforms/posix/src/px4/generic/generic/include/px4_arch/micro_hal.h:36: new blank line at EOF.
platforms/posix/src/px4/generic/generic/tone_alarm/ToneAlarmInterface.cpp:60: new blank line at EOF.
platforms/qurt/src/px4/CMakeLists.txt:37: new blank line at EOF.
platforms/qurt/src/px4/common/get_commands.h:48: new blank line at EOF.
platforms/qurt/src/px4/common/include/px4_platform/micro_hal.h:36: new blank line at EOF.
platforms/qurt/src/px4/generic/CMakeLists.txt:36: new blank line at EOF.
platforms/qurt/src/px4/generic/generic/CMakeLists.txt:33: new blank line at EOF.
platforms/qurt/src/px4/generic/generic/include/px4_arch/micro_hal.h:36: new blank line at EOF.
src/drivers/driver_framework_wrapper/df_ms5611_wrapper/CMakeLists.txt:46: new blank line at EOF.
src/drivers/driver_framework_wrapper/df_trone_wrapper/CMakeLists.txt:46: new blank line at EOF.
src/drivers/drv_adc.h:91: new blank line at EOF.
src/lib/mixer_module/CMakeLists.txt:35: new blank line at EOF.
src/lib/mixer_module/mixer_module.cpp:527: new blank line at EOF.
src/lib/mixer_module/mixer_module.hpp:240: new blank line at EOF.
src/lib/output_limit/output_limit.h:80: new blank line at EOF.
src/platforms/px4_init.h:54: new blank line at EOF.
src/platforms/px4_micro_hal.h:41: new blank line at EOF.
```

mcsauder commented:
Rebased with current master and changes from #12934 pulled into this PR. Bench tests look as they should:
![image](https://user-images.githubusercontent.com/2497951/64713998-1b900500-d47b-11e9-9f3f-f191a0d65cda.png)

mcsauder commented:
Fixed two copy/paste typos, squashed, rebased, and retested.  There is more work I'd like to accomplish in this driver, but I am happy with this PR right now. Let me know if you have any questions on this PR.

![image](https://user-images.githubusercontent.com/2497951/64914900-5b274d00-d718-11e9-8f71-6788b9c2ea64.png)

mcsauder commented:
Merged current master and rebased, no code changes.

julianoes commented on src/drivers/distance_sensor/ll40ls/ll40ls.cpp:
```diff
@@ -89,159 +63,227 @@ namespace ll40ls
+
+The sensor/driver must be enabled using the parameter SENS_EN_LL40LS.
+
+Setup/usage information: https://docs.px4.io/v1.9.0/en/sensor/lidar_lite.htmls
```
I would probably not reference the v1.9.0 version here but just use `master`.

mcsauder replied:
Nice catch! updated. Thanks!

julianoes commented on src/drivers/distance_sensor/ll40ls/LidarLite.h:
```diff
@@ -44,22 +44,24 @@
-// maximum time to wait for a conversion to complete.
-#define LL40LS_CONVERSION_TIMEOUT 100*1000UL /* 100ms */
+// Maximum time to wait for a conversion to complete.
+#define LL40LS_CONVERSION_TIMEOUT 100_ms
```
While you're at it you could make these `static constexpr auto`, right?

mcsauder replied:
Done! Thanks!

TSC21 commented on src/drivers/distance_sensor/ll40ls/LidarLiteI2C.h:
```diff
@@ -42,114 +42,125 @@
+static constexpr int LL40LS_SIGNAL_STRENGTH_MIN_V3HP  = 70;  /* Min signal strength for V3HP */
+static constexpr int LL40LS_SIGNAL_STRENGTH_MAX_V3HP  = 255; /* Max signal strength for V3HP */
+
+static constexpr int LL40LS_SIGNAL_STRENGTH_LOW = 24;  /* Minimum relative signal strength for a valid measurement */
```
Could we add the same indentation here as above?

mcsauder replied:
I shortened the comments just a little so I could do that and get past the format checks (indenting to the same as the previous lines pushed these lines over 120 characters on the lines).  Thanks a ton for your review and feedback @TSC21!

mcsauder commented:
Thanks @TSC21 !  I appreciate your review!
